hack/test/e2e-embedded.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+set -eoux pipefail
+
+source ./hack/test/e2e.sh
+
+PROVISIONER=qemu
+CLUSTER_NAME=e2e-embedded
+
+NODE="172.20.3.2"
+
+function build_iso_embedded {
+  # build the ISO with embedded config
+  cp hack/test/patches/watchdog.yaml ${ARTIFACTS}/embedded.yaml
+  make image-iso IMAGER_ARGS="--embedded-config-path=/out/embedded.yaml" PLATFORM=linux/amd64
+}
+
+function create_cluster {
+  "${TALOSCTL}" cluster create \
+    --provisioner="${PROVISIONER}" \
+    --name="${CLUSTER_NAME}" \
+    --iso-path=${ARTIFACTS}/metal-amd64.iso \
+    --controlplanes=1 \
+    --workers=0 \
+    --mtu=1430 \
+    --memory=2048 \
+    --cpus=2.0 \
+    --cidr=172.20.3.0/24 \
+    --skip-injecting-config \
+    --wait=false \
+    --cni-bundle-url=${ARTIFACTS}/talosctl-cni-bundle-'${ARCH}'.tar.gz
+}
+
+function destroy_cluster() {
+  "${TALOSCTL}" cluster destroy \
+    --name "${CLUSTER_NAME}" \
+    --provisioner "${PROVISIONER}" \
+    --save-cluster-logs-archive-path="/tmp/logs-${CLUSTER_NAME}.tar.gz" \
+    --save-support-archive-path="/tmp/support-${CLUSTER_NAME}.zip"
+}
+
+trap destroy_cluster SIGINT EXIT
+
+build_iso_embedded
+create_cluster
+sleep 5
```

Comment:
Does this need this wait? It can perhaps be handled by the for loop below as well